### DISCUSSION TITLE
Why does consult-ripgrep return duplicate results when passing `-- PATH` ?

### DISCUSSION THREAD

**Discussion by ldeck** [minad/consult#1003] [Q&A] [answered]
I'm seeing an interesting behaviour

If I run `project-find-regexp` (which has :override advice: `consult-ripgrep`) with `#setValue -- src/main/java` I'm presented with duplicate results.

If I run the same with `#setValue#src/main/java` I'm presented with unique results but I would presume it's slightly less efficient.

Does anyone know why the former one behaves that way? Doing the same on a shell works fine. i.e., `rg setValue src/main/java`.

Btw, my `consult-ripgrep-args` has value "rg --null --line-buffered --color=never --max-columns=1000 --path-separator /   --smart-case --no-heading --with-filename --line-number --search-zip" if that's of any relevance, which appears to be the default defined in `consult.el`.

**Comment by minad** [comment, marked as answer]
This problem has come up before. I think the issue is that the path passed to ripgrep is somehow duplicated, for example if `consult-ripgrep-args` already includes "." as argument and if you add an additional path. Check the ` *consult-async*` buffer for the list of options passed to rg.

**Reply by ldeck** [reply]
Thanks @minad for that!

So, yes the `"."` is being appended:
`consult--async-process started ("rg" "--null" "--line-buffered" "--color=never" "--max-columns=1000" "--path-separator" "/" "--smart-case" "--no-heading" "--with-filename" "--line-number" "--search-zip" "-P" "-e" "setValue" "src/main/java" ".")`

Is there a switch that controls the appending of `"."` that you know of?

**Reply by ldeck** [reply]
Okay it seems to be the behaviour of the `consult--directory-prompt` function to append a default directory and the autoload `consult-ripgrep` takes params `(&optional dir initial)`.

I don't grok how that `dir` argument gets populated.

Looking at the docs for `project-find-regexp` it states:
```
 (defun project-find-regexp (regexp)
  "Find all matches for REGEXP in the current project's roots.
With \\[universal-argument] prefix, you can specify the directory
to search in, and the file name pattern to search for...
```

So it would appear that being able to specify a prefix argument might do the trick (not sure if that's possible) or defining a custom interactive function that calls `(consult-ripgrep with-some-custom-dir "")`.

**Reply by ldeck** [reply]
@minad is this the kind of function that would do the trick? 

```
(defun ld/consult-ripgrep-in-project-dir ()
  "Use `project-find-dir' to select a directory and then call `consult-ripgrep' to find files."
  (interactive)
  (let ((selected-dir (project-find-dir))
        (initial ""))
    (if selected-dir
        (consult-ripgrep selected-dir initial)
      (message "No project directory found."))))

```

I'm getting an error, however, after selecting the directory:
```
apply: Autoloading file /nix/store/ryhysci9nkcfx8smx47g4jbsicqjcvv3-emacs-packages-deps/share/emacs/site-lisp/elpa/consult-20231211.2102/consult.elc failed to define function consult-completing-read-multiple
```

**Reply by ldeck** [reply]
Doh, apologies @minad, I see consult-crm had been deprecated and removed.

I also want to avoid dired. I've managed to cobble together the following function that works perfectly.

```
(defun ld/consult-ripgrep-in-project-subdir ()
  "Search for a pattern using ripgrep in a subdirectory of the current project."
  (interactive)
  (let* ((project-root (project-root (project-current)))
         (subdirs (mapcar (lambda (dir) (file-relative-name dir project-root))
                          (seq-filter 'file-directory-p
                                      (directory-files-recursively project-root "^[^.].*" t)))))
    (if subdirs
        (let ((subdir (completing-read "Select a subdirectory: " subdirs)))
          (if subdir
              (let* ((default-directory (expand-file-name subdir project-root)))
                (consult-ripgrep default-directory))
            (message "No subdirectory selected.")))
      (message "No subdirectories found in the project."))))
```
